packages/smooth_app/lib/pages/product/product_attributes_page.dart

```diff
@@ -0,0 +1,344 @@
+
+const String splitChar = ':';
+
+class ProductAttributesPage extends StatefulWidget {
+  const ProductAttributesPage(this.product);
+
+  final Product product;
+
+  @override
+  State<ProductAttributesPage> createState() => _ProductAttributesPageState();
+}
+
+class _ProductAttributesPageState extends State<ProductAttributesPage>
+    with UpToDateMixin {
+  final ScrollController _controller = ScrollController();
+  // OrderedNutrientsCache? cache;
+
+  @override
+  void initState() {
+    super.initState();
+    initUpToDate(widget.product, context.read<LocalDatabase>());
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final AppLocalizations appLocalizations = AppLocalizations.of(context);
+    context.watch<LocalDatabase>();
+    refreshUpToDate();
+    final String productName = getProductName(
+      upToDateProduct,
+      appLocalizations,
+    );
+    final String productBrand =
+        getProductBrands(upToDateProduct, appLocalizations);
+
+    return SmoothScaffold(
+      appBar: ProductAppBar(
+        barcode: barcode,
+        barcodeVisibleInAppbar: true,
+        productBrand: productBrand,
+        productName: productName,
+      ),
+      body: RefreshIndicator(
+        onRefresh: () async => ProductRefresher().fetchAndRefresh(
+          barcode: barcode,
+          widget: this,
+        ),
+        child: Scrollbar(
+          controller: _controller,
+          child: ListView(
+            controller: _controller,
+            physics: const AlwaysScrollableScrollPhysics(),
+            children: <Widget>[
+              AttributeItems(
+                helper: SimpleInputPageCategoryHelper(),
+                product: upToDateProduct,
+              ),
+              AttributeItems(
+                helper: SimpleInputPageLabelHelper(),
+                product: upToDateProduct,
+              ),
+              NutritionFacts(
+                product: upToDateProduct,
+              ),
+              Ingredients(
+                product: upToDateProduct,
+              ),
+              AttributeItems(
+                helper: SimpleInputPageCountryHelper(),
+                product: upToDateProduct,
+              ),
+              AttributeItems(
+                helper: SimpleInputPageStoreHelper(),
+                product: upToDateProduct,
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+}
+
+class NutritionFacts extends StatefulWidget {
+  const NutritionFacts({required this.product});
+
+  final Product product;
+  @override
+  State<NutritionFacts> createState() => _NutritionFactsState();
+}
+
+class _NutritionFactsState extends State<NutritionFacts> {
+  List<String> allTerms = <String>[];
+
+  @override
+  Widget build(BuildContext context) {
+    widget.product.knowledgePanels?.panelIdToPanelMap['nutrition_facts_table']
```

Comment:
I hope there's an easier way to get nutrition facts.

Reply:
+ it would be better if this would be computed only once and not for every build

Reply:
yeah! `product.nutriments`
But problem with that is the way its configured! so thinking that the names can have any symbol i just used this method which gives a map of key value pair !
So what should be done ?

Reply:
@Sudhanva-Nadiger We should ignore the knowledge panels here, and loop on the nutrients (ordered according to the country), displaying whatever is populated.